src/llmq/core_write.cpp

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2018-2025 The Dash Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <llmq/commitment.h>
+#include <llmq/signing.h>
+#include <llmq/snapshot.h>
+
+#include <core_io.h>
+#include <rpc/util.h>
+
+#include <univalue.h>
+
+#include <map>
+#include <string>
+
+namespace llmq {
+RPCResult CFinalCommitment::GetJsonHelp(const std::string& key, bool optional)
```

Comment:
nit: I don't really like RPC code be spread all over codebase in different subdirectories, `src/llmq/core_write.cpp` is not the best place to have it, but I have no better idea at the moment.

Reply:
Yeah, I don't like obviously RPC code living outside the `rpc/` directory but our problem is twofold

* We need the sources to be a part of `libbitcoin_common` (the whole reason we've made `{evo,llmq}/core_write.cpp`)

* We need the help text to be right next to the definition so they don't go out of sync (but even this rule had to be broken as two `ToJson()` definitions use txindex, which is not a part of `libbitcoin_common` and moving `GetJsonInfo()` to those source files creates very ugly circular dependencies)

What makes this acceptable is that `RPCHelpArg` is a relatively uncomplicated structure that is mostly defined in the header.